### PR TITLE
Rocks Randomly Placed In Level

### PR DESCRIPTION
**Pull request by alexottoboni**
Places two kinds of rocks in the level. 

Bug (maybe related?): Some moving platforms move along the Z and if you land on them they mess up your Z position.

### CODE REVIEW

File: src/RhythmRunner.cpp [hunk 1]
```diff
@@ -24,6 +24,11 @@
 
 #define MUSIC "music/2.wav"
 
+std::shared_ptr<MenuState> menu_state;
+std::shared_ptr<GameState> game_state;
+
+GameUpdater game_updater;
+
```
Comment (josepharhar) [inline src/RhythmRunner.cpp]: ?

File: src/RhythmRunner.cpp [hunk 2]
```diff
@@ -130,13 +135,10 @@ int main(int argc, char** argv) {
           } else {
             level_generator = new LevelGenerator(menu_state->GetMusicPath());
           }
-          game_state = std::make_shared<GameState>(
-              level_generator->generateLevel(), std::make_shared<GameCamera>(),
-              std::make_shared<Player>(), std::make_shared<Sky>());
-          // Only Video texture we have right now
-          std::shared_ptr<VideoTexture> vid = std::make_shared<VideoTexture>(
-              std::string(ASSET_DIR) + "/textures/sky");
-          game_state->AddVideoTexture("sky", vid);
```
Comment (josepharhar) [inline src/RhythmRunner.cpp]: Why delete this?

File: src/game_state/DroppingPlatform.cpp [hunk 3]
```diff
@@ -33,6 +33,8 @@ DroppingPlatform::DroppingPlatform(glm::vec3 position)
 DroppingPlatform::DroppingPlatform(glm::vec3 position, glm::vec3 scale)
     : Obstacle(DroppingPlatform::shape), originalPosition(position) {
   this->position = position;
+  this->scale = glm::vec3(3, .5, 7);
+  this->model = shape;
```
Comment (josepharhar) [inline src/game_state/DroppingPlatform.cpp]: ?

File: src/game_state/Platform.cpp [hunk 4]
```diff
@@ -18,7 +18,7 @@ Platform::Platform(glm::vec3 position) : Obstacle(Platform::platform) {
 Platform::Platform(glm::vec3 position, glm::vec3 scale)
     : Obstacle(Platform::platform) {
   this->position = position;
-  this->scale = scale;
+  this->scale = glm::vec3(4, .5, 7);
```
Comment (josepharhar) [inline src/game_state/Platform.cpp]: ?

File: src/generator/LevelGenerator.cpp [hunk 5]
```diff
@@ -181,6 +186,29 @@ LevelGenerator::Generate() {
           }
           objs->push_back(std::make_shared<gameobject::Platform>(
               glm::vec3(xPos, yPos, zPos), glm::vec3(power, .5f, 7.0f)));
+          if ((rand() % 2) + 1 == 1) {
```
Comment (josepharhar) [inline src/generator/LevelGenerator.cpp]: I might be late saying this if there are already uses of rand() in the level generator, but I don't think that you should get a different level each time you play the same song

Comment (Matthew-Stewart) [reply src/generator/LevelGenerator.cpp]: This is more of a design decision, but I think as long as the difficulty doesn't vary too greatly (and we don't make impossible levels), making the level have variable elements makes it more exciting. just my opinion though. team meeting? 👍

File: src/game_state/DroppingPlatform.cpp (hunk 3 above)
Comment (bbeck13) [reply src/game_state/DroppingPlatform.cpp]: I fixed this in the main branch this branch needs to be rebased

File: src/generator/LevelGenerator.cpp (hunk 5 above)
Comment (bbeck13) [reply src/generator/LevelGenerator.cpp]: I can change this I'll put out a pr for some more level generation stuff

**Comment by bbeck13** [comment]
CR 👍 QA 👍

**Pull request merged**